### DISCUSSION TITLE
commafeed.com current configuration

### DISCUSSION THREAD

**Discussion by Athou** [Athou/commafeed#1567] [Announcements]
In order to be as transparent as possible, here's the current configuration of the commafeed.com public instance.

## Feed fetching

- Feeds are fetched every 10 minutes to 24 hours. The actual interval is based on the average time between entries in the feed.
- Users that have not interacted with commafeed.com either via the web client or via the API for 30 days will not have their feeds refreshed until they interact with the website again.
- There is a 30 minutes cooldown on the "Fetch all my feeds now" action.
- Push notifications on new entries are disabled

## Feed retention

- Feed entries are kept for 365 days.
- Maximum capacity is 500 entries per feed. When a feed reaches that limit, old entries are removed even if they are not older than 365 days.
- Starred entries are subject to those rules.
- Feeds larger than 5Mb are ignored to prevent memory issues while parsing.
- Unread statuses are kept for 30 days. Entries older than 30 days are automatically considered read.
- Accounts are limited to 1000 feeds.

Some of those settings may seem inconvenient but they are required to keep a free instance available for everyone.
There are no guarantees that those settings will never change, the post will be updated if they do.

With that in mind, you can decide if those settings are fine for your usage of CommaFeed or if you'd rather roll your own instance 🙂

**Comment by peterdemin** [comment]
Hi, [Dan Luu](https://danluu.com/) likes to write long posts and has a lot of them. His atom.xml is a bit over 6 MB. Is there a way to bump the 5MB limit via config/settings?

**Reply by Athou** [reply]
Of course, see `commafeed.http-client.max-response-size` [here](https://athou.github.io/commafeed/documentation/). You'll have to self-host an instance of CommaFeed though.

**Reply by peterdemin** [reply]
Thanks for the quick reply! I updated my SystemD unit to pass it as an env var:
https://peter.demin.dev/12_articles/71-rss-feed.html#:~:text=COMMAFEED_HTTP_CLIENT_MAX_RESPONSE_SIZE
Now I need to wait an hour until the next refresh 🤞🏻

**Reply by Athou** [reply]
I read the page and I have some comments if you don't mind 😊
- Is the WebSocket connection working? I remember nginx requiring additional configuration to proxy pass ws connections (https://nginx.org/en/docs/http/websocket.html)
- If you generate an API key on the profile page, you can download your OPML file via the REST API (https://www.commafeed.com/api-documentation/#/Feeds/get_rest_feed_export) and use that to backup automatically at regular intervals. Just append `?apiKey=<your key>` at the end of the url to authenticate yourself.
- By enabling registrations and creating an account four yourself, I don't think you disabled the default admin/admin user. I suggest you log in with the admin account, give your other account admin rights, then delete the default admin account when logged in with your personal account. People usually just change the default admin password right after installing CommaFeed. I should probably create a landing page to do all that during the initial setup.

**Reply by peterdemin** [reply]
Thank you, much appreciated! I updated the nginx config (and verified that `/ws` no longer fails with `404`), fixed the account registration instructions, and automated the OPML export.

**Comment by AlFirous** [comment]
This is very generous, even compared to Inoreader or Feedly. I need to test it out before self-hosting. Thanks for that.
